File: examples/quickstart/quickstart.py
"""AGR Python quickstart flow."""

from __future__ import annotations

import os
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[2] / "packages" / "agr-sdk-python"))
from agr.client import AGRClient


def main() -> None:
    client = AGRClient(
        api_key=os.environ.get("AGR_API_KEY"),
        base_url=os.environ.get("AGR_BASE_URL", "http://localhost:8000"),
    )

    agent = client.register_agent("example-agent", {"framework": "quickstart"})
    print("Registered agent:", agent.get("agent_id"))

    allow_result = client.evaluate(
        "example-agent",
        "read_docs",
        "getting-started",
        {"environment": "development", "risk_level": "low"},
    )
    print("Low-risk decision:", allow_result.decision, "-", allow_result.reason)

    approval_result = client.evaluate(
        "example-agent",
        "deploy",
        "checkout-service",
        {"environment": "production"},
    )
    print("Sensitive decision:", approval_result.decision, "-", approval_result.reason)

    if approval_result.approval_id:
        try:
            approved = client.wait_for_approval(approval_result.approval_id, timeout=5.0)
            print("Approval resolved:", "approved" if approved else "rejected")
        except TimeoutError:
            print("Approval still pending after 5 seconds. Exiting gracefully.")

    client.close()


if __name__ == "__main__":
    main()

File: examples/quickstart/quickstart.ts
import { AGRClient } from "../../packages/agr-sdk-ts/src/client.js";

async function main(): Promise<void> {
  const client = new AGRClient({
    apiKey: process.env["AGR_API_KEY"],
    baseUrl: process.env["AGR_BASE_URL"] ?? "http://localhost:8000",
  });

  const agent = await client.registerAgent("example-agent", { framework: "quickstart" });
  console.log("Registered agent:", agent.agent_id);

  const allowResult = await client.evaluate(
    "example-agent",
    "read_docs",
    "getting-started",
    { environment: "development", risk_level: "low" },
  );
  console.log("Low-risk decision:", allowResult.decision, "-", allowResult.reason);

  const approvalResult = await client.evaluate(
    "example-agent",
    "deploy",
    "checkout-service",
    { environment: "production" },
  );
  console.log("Sensitive decision:", approvalResult.decision, "-", approvalResult.reason);

  if (approvalResult.approvalId) {
    try {
      const approved = await client.waitForApproval(approvalResult.approvalId, { timeout: 5000 });
      console.log("Approval resolved:", approved ? "approved" : "rejected");
    } catch (error) {
      console.log("Approval still pending after 5 seconds. Exiting gracefully.");
    }
  }
}

void main();
